### DISCUSSION TITLE
Questions sur l'API

### DISCUSSION THREAD

**Discussion by julj** [PnX-SI/GeoNature#2979] [General]
Bonjour,

nous travaillons actuellement sur une plateforme d'identification automatique d'espèces et souhaitons pouvoir envoyer nos données sur une instance Geonature.

Nous utilisons actuellement l'API (non documentée) utilisée dans Geonature pour créer les objets nécessaires: Cadre d'acquisition, Jeu de données, Relevé, Taxon.

J'ai lu dans un commentaire de ticket qu'il était préférable de créer directement un objet "synthèse", et j'ai vu qu'[une PR](https://github.com/PnX-SI/GeoNature/pull/2354) était en cours.

Est-ce bien toujours la méthode recommandée pour créer de large quantité de données dans Geonature ?

Deux autres questions (sur lesquelles nous pourrions contribuer):
- Une API publique est-elle prévue ?
- Avez-vous une stratégie pour l'intégration des données issues de détection automatique par IA ? Je pense en particulier à la possibilité de pouvoir avoir une IA comme identifieur de l'espèce.

Bonne journée,

Julien Ricard
[Terroiko](https://www.terroiko.fr)

**Comment by camillemonchicourt** [comment]
En effet, actuellement l'API des modules GeoNature est considérée comme spécifique et interne à GeoNature et n'est pas prévue ni garantie pour des interactions avec des outils tiers.

Cela n'empêche pas certains de l'utiliser, mais elle n'a pas été prévue pour cela, n'est pas standardisée, pas documentée et pas versionnée. Elle peut évoluer pour les besoins internes de GeoNature.

La seule API qui est prévue actuellement pour des interactions sortantes avec des outils tiers est celle du module Export.

Il y a clairement un intérêt et une volonté que l'API de GeoNature soit utilisable par des outils tiers, en GET et en POST, stable, standard (REST), documentée, versionnée, mais nous n'avons pas encore pu avancer sur ce sujet.

La priorité qui a été identifiée est en effet l'ajout du API REST (Get / Post) sur le module Synthèse car il est central et c'est là qu'il y a le plus d'enjeu de pouvoir intégrer des données depuis des outils tiers. Il faudrait rediscuter de la solution et architecture technique retenue et terminer la PR que vous avez identifié.

Le module Occtax est quant à lui plutôt penser comme un module de saisie, donc selon votre contexte et votre besoin, il est plutôt encourager d'alimenter la Synthèse (qui agrège toutes les observations de taxons depuis différentes sources) plutôt qu'Occtax.

Donc actuellement, l'utilisation des API POST internes à GeoNature est possible et certains le font, mais il n'est pas garanti ni supporté.
On utilise d'ailleurs l'API du module Occtax (notamment) dans Occtax-mobile (https://github.com/PnX-SI/gn_mobile_occtax).

Mais il faudrait donc avancer sur le sujet, notamment au niveau de la Synthèse, pour pouvoir supporter cet usage, le documenter et le maintenir.

--------------------------

Pour l'API publique, voir au niveau du module EXPORT (https://github.com/PnX-SI/gn_module_export), fait pour ça et entièrement customisable et dynamique.

--------------------------

Concernant l'aide à l'identification d'espèces par IA, le sujet a été évoqué quelques fois, sans aller plus loin.
Mais l'idée de pouvoir interagir avec une API (comme Plantnet ou autre) a été évoqué plusieurs, notamment au niveau d'Occtax-mobile : https://github.com/PnX-SI/gn_mobile_occtax/issues/234
